x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -124,6 +124,12 @@ func (r *metricsRequester) Metrics(ctx context.Context, serviceName string, alig
 		}
 	}
 
+	start := time.Now()
+	defer func() {
+		totalTime := time.Since(start)
+		r.logger.Debugf("Total time taken for all ListTimeSeries requests: %s and config period: %s", totalTime, r.config.period.AsDuration())
+	}()
```

Comment:
Same as above,

```suggestion
	defer func() {
		r.logger.Debugf("Total time taken for all ListTimeSeries requests: %s and config period: %s", time.Since(start), r.config.period.AsDuration())
	}()
```